### PR TITLE
Update skipIfNoCuda decorator/Force GPU tests run in GPU CIs

### PR DESCRIPTION
**Pull request by salilsdesai**
In response to [https://github.com/pytorch/audio/issues/1552](https://github.com/pytorch/audio/issues/1552)

- Update the skipIfNoCuda decorator to raise an error if the TORCHAUDIO_TEST_FORCE_CUDA variable is set but CUDA is not available
- Update CI to use TORCHAUDIO_TEST_FORCE_CUDA in gpu unittests

### CODE REVIEW

**Comment by mthrok** [comment]
Hi @salilsdesai 

Thanks for working on this. Instead of changing `config.yml` directly, you need to apply the change to `config.yml.in` and use `regenerate.py` to update `config.yml`. We do this because we have to define the same CI jobs to different Python / CUDA version.

File: .circleci/config.yml.in [hunk 1]
```diff
@@ -465,7 +465,7 @@ jobs:
           command: docker run -t --gpus all -e UPLOAD_CHANNEL -e CONDA_CHANNEL_FLAGS -v $PWD:$PWD -w $PWD "${image_name}" .circleci/unittest/linux/scripts/install.sh
       - run:
           name: Run tests
-          command: docker run -t --gpus all -v $PWD:$PWD -w $PWD "${image_name}" .circleci/unittest/linux/scripts/run_test.sh
+          command: docker run -t --gpus all -v $PWD:$PWD -w $PWD "${image_name}" -e "TORCHAUDIO_TEST_FORCE_CUDA=1" .circleci/unittest/linux/scripts/run_test.sh
```
Comment (mthrok) [inline .circleci/config.yml.in:468]: `-e` has to be before `"${image_name}"`

**Comment by mthrok** [comment]
Looks like our CI does have GPU but the firmware is too old. I see  a warning

```
  c:\users\circleci\project\env\lib\site-packages\torch\cuda\__init__.py:52: UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 10010). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at  ..\c10\cuda\CUDAFunctions.cpp:115.)
```

https://app.circleci.com/pipelines/github/pytorch/audio/6271/workflows/66eeb5f8-1841-4b97-a0b2-9c7a36adf4a5/jobs/232928/steps

Let me see what's the alternative way to install PyTorch.

**Comment by mthrok** [comment]
@salilsdesai Can you leave the Windows as-is for now? The Windows CI situation is somewhat complicated to resolve, so we will tackle later. The you are adding is variable so let's merge it first.

**Comment by salilsdesai** [comment]
> @salilsdesai Can you leave the Windows as-is for now? The Windows CI situation is somewhat complicated to resolve, so we will tackle later. The you are adding is variable so let's merge it first.

Oh, you mean I should revert the windows gpu test, right? I'll do that now then.

**Comment by mthrok** [comment]
> > @salilsdesai Can you leave the Windows as-is for now? The Windows CI situation is somewhat complicated to resolve, so we will tackle later. The you are adding is variable so let's merge it first.
> 
> Oh, you mean I should revert the windows gpu test, right? I'll do that now then.

Yes, exactly. Thanks for updating it.

**Pull request merged**